gateway/1.4/modules/ROOT/pages/flex-gateway-k8-high-availability.adoc

```diff
@@ -0,0 +1,286 @@
+--reuse-values \
+-f <your-autoscaling-configuration-yaml>
+----
++
+.Example:
+[source,helm]
+----
+helm -n gateway upgrade \
+ingress flex-gateway/flex-gateway \
+--reuse-values \
+-f autoscaling.yaml
+----
++
+When successful, the command prints output similar to this example: 
++
+----
+Release "ingress" has been upgraded. Happy Helming!
+NAME: ingress
+LAST DEPLOYED: Tue Apr 18 15:46:38 2023
+NAMESPACE: gateway
+STATUS: deployed
+REVISION: 33
+TEST SUITE: None
+----
+
+. If the Helm chart is not installed, run the following command to apply the your autoscaling configuration:
++
+.Syntax:
+----
+helm -n <namespace> \
+upgrade -i --create-namespace \
+<release-name> <helm-repository-name>/<helm-chart-name> \
+--set-file registration.content=<registration file> \
+-f <your-autoscaling-configuration-yaml>
+----
++
+.Example:
+[source,helm]
+----
+helm -n gateway \
+upgrade -i --create-namespace \
+ingress flex-gateway/flex-gateway \
+--set-file registration.content=registration.yaml \
+-f autoscaling.yaml
+----
++
+When successful, the command prints output similar to this example: 
++
+----
+NAME: ingress
+LAST DEPLOYED: Tue Apr 18 15:50:29 2023
+NAMESPACE: gateway
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+----
+
+. Verify your autoscaling configuration by running the following command:
++
+.Syntax:
+[source, kubectl]
+----
+kubectl get hpa -n <namespace>
+----
++
+.Example:
+[source, kubectl]
+----
+kubectl get hpa -n gateway
+----
++
+The command prints output similar to this example: 
++
+----
+NAME     REFERENCE           TARGETS          MINPODS  MAXPODS  REPLICAS  AGE
+ingress  Deployment/ingress  66%/50%, 5%/50%  2        11       2         59s
+----
+//TODO: TIME PERMITTING, IT WOULD BE GOOD TO ADD SOME TROUBLESHOOTING INFO
+
+[[autoscaling-parameters]]
+=== Autoscaling Parameters
+
+To use autoscaling, enable Horizontal Pod Autoscaler (HPA), and configure other Helm chart settings to define autoscaling behavior. Autoscaling parameters in the Helm chart follow.
+
+[%header,cols="1a,2a"]
+|===
+| Parameter | Description
+| `autoscaling.enabled` | Boolean that indicates whether the Horizontal Pod Autoscaler (HPA) is enabled. Defaults to `false`.
+| `autoscaling.minReplicas` | The minimum number of replicas that the HPA scaler is allowed to create. Defaults to `2`.
+| `autoscaling.maxReplicas` | The maximum number of replicas that the HPA scaler is allowed to create. Defaults to `11`.
+| `autoscaling.targetCPUUtilizationPercentage` | A resource metric that sets the average CPU usage percentage of all deployed Pods. Defaults to `50`.
+| `autoscaling.targetMemoryUtilizationPercentage` | A string representing the average memory usage percentage of all deployed Pods. Defaults to `nil`.
+| `autoscaling.extraMetrics` | A list of custom Pod configuration metrics that enables you to collect and monitor application-specific metric data. Defaults to `[]`. Define the metrics with nested parameters, for example:
```

Comment:
Does the list enable you or the metrics? If list, it's fine. If metrics, "enable".

Reply:
Hmm, the list of settings are applied to the Pod, and if the Pod runs with those settings, then you'll be able to get the data. Let me try to rework it a bit.

Reply:
revised